src/cluster.c

```diff
@@ -1935,6 +1935,15 @@ int clusterProcessPacket(clusterLink *link) {
          * of the message type. */
         if (!sender && type == CLUSTERMSG_TYPE_MEET)
             clusterProcessGossipSection(hdr,link);
+	    
+	if (sender && type == CLUSTERMSG_TYPE_PING && link->node == NULL){
+            link->node = sender;
+            if(sender->link_from) {
+                sender->link_from->node = NULL;
+                freeClusterLink(sender->link_from);
+            }
+            sender->link_from = link;
+        }
```

Comment:
There is some weird spacing here.
```suggestion
        if (sender && type == CLUSTERMSG_TYPE_PING && link->node == NULL) {
            link->node = sender;
            if (sender->link_from) {
                sender->link_from->node = NULL;
                freeClusterLink(sender->link_from);
            }
            sender->link_from = link;
        }
```